derive/src/decode.rs

```diff
@@ -72,7 +72,12 @@ pub fn quote(
 
 				quote_spanned! { v.span() =>
 					__codec_x_edqy if __codec_x_edqy == #index as ::core::primitive::u8 => {
-						#create
+						// NOTE: This lambda is necessary to work around an upstream bug
+						// where each extra branch results in excessive stack usage:
+						//   https://github.com/rust-lang/rust/issues/34283
+						return (move || {
```

Comment:
I haven't checked, but I highly doubt it will. And we're not serializing/deserializing that much anyway (if we would I would have worked on optimizing `parity-scale-codec` more, as it is relatively slow, e.g. [compare it to my `speedy` here](https://github.com/djkoloski/rust_serialization_benchmark#log))

If you want maybe we could run the benchmarks once I'll make the PR to pull it into substrate?

Reply:
I run the criterion before and after and there is no regression, so should be good.